### DISCUSSION TITLE
Sorting by indirect relation fields

### DISCUSSION THREAD

**Discussion by hlanderdev** [pocketbase/pocketbase#1429] [Q&A] [answered]
Is it possible to sort by indirect relation fields? For example:

```
?expand=comments(post).user&sort=+comments(post).user.name
```

**Comment by ganigeorgiev** [comment, marked as answer]
No, currently filtering and sorting of indirect relations is not supported (it is in my todo to research it, but it is left out of scope for now).

**Reply by hlanderdev** [reply]
Ok, no problem. Thank you!

**Comment by danilomsilva** [comment]
Hi there,
is that supported now? If so, could you please share any example?
Thank you

**Reply by ganigeorgiev** [reply]
No it is not supported and it is pushed back for now.

**Reply by starredev** [reply]
I'm also waiting for this, are there any workarounds yet?

**Reply by othema** [reply]
+1

**Reply by ganigeorgiev** [reply]
_Side-note: Please don't spam the discussion. If this feature request is important for you, try to provide more details about your use case as there could be other solution for it._

**Reply by starredev** [reply]
I've created  dynamic navigation menu with the following collection structure.

```
await db.collection('navigation').getFullList({
                fields: 'id, title, route, expand.categories.title, expand.categories.expand.items.title, expand.categories.expand.items.route',
                expand: 'categories.items',
                sort: '+order'
            })
```
I want to sort top navigation (menu..), categories(submenu) and item

Legacy Menu     PON Menu   etc
  submenu            submenu
    item                   item
    item                   item

**Reply by ganigeorgiev** [reply]
@raizdev Hm, I'm not entirely sure that I understand the above example, but note that the back-relation sort will be applied only on the main records (in your case "navigation"), not on the expanded relation items (aka. it will be similar to how it currently works with the direct relations).

If you need sorting for the relation items you can send a separate request to the "items" collection.
To retrieve only the "items" for your previously fetched "navigation", you can extract the relation ids from the "navigations" records and construct a filter for the "items" request like `id=id1||id=id2||id=id3||...` (_aka. similar to an `IN` SQL query_).

**Comment by Karnak19** [comment]
Also interested about this, I have something like:
- `builds` collection
- `builds_votes` collection

Only the `build` is referenced in `builds_votes` to avoid having a big array of `builds_votes` when fetching `builds`.

Currently i am able to `expand: 'builds_votes(build)'`, but I would love to be able to `sort: +builds_votes(build):length` too !

**Reply by ganigeorgiev** [reply]
No, sorry. It is out of the v1 scope for now.

**Reply by ganigeorgiev** [reply]
I forgot to mention that specifically for your use case an eventual alternative could be to create a helper View collection that will aggregate the count per build and then use it in your sort expression, something like:

```sql
-- buildsVotesCounts
SELECT
  (ROW_NUMBER() OVER()) as id,
  build,
  count(*) as total
FROM builds_votes
GROUP BY build
```

`sort: buildsVotesCounts_via_build.total`

_edit: replaced the old back/indirect relation syntax with the new `_via_`._

**Reply by Karnak19** [reply]
i'll dig this way, thank you for the hint
